src/main/java/it/gov/pagopa/pu/organization/repository/OrganizationRepository.java

```diff
@@ -14,4 +19,15 @@ public interface OrganizationRepository extends JpaRepository<Organization, Long
   Optional<Organization> findByOrgFiscalCode(String orgFiscalCode);
   List<Organization> findByBrokerId(Long brokerId);
 
+  @Modifying
+  @Transactional
+  @RestResource(exported = false)
+  @Query("UPDATE Organization o SET " +
+    "o.ioApiKey = CASE WHEN :typeEnum = 'IO' THEN :apiKey ELSE o.ioApiKey END, " +
+    "o.sendApiKey = CASE WHEN :typeEnum = 'SEND' THEN :apiKey ELSE o.sendApiKey END " +
+    "WHERE o.id = :organizationId")
+  void updateApiKeyByType(@Param("organizationId") Long organizationId,
```

Comment:
just write 2 separate update query and call the specific one based on the enum.
This approach is difficult to debug